pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -584,22 +587,19 @@ func (vs *VSphere) DisksAreAttached(volPaths []string, nodeName k8stypes.NodeNam
 		}
 		vm, err := vs.getVMByName(ctx, nodeName)
 		if err != nil {
+			if vclib.IsNotFound(err) {
+				glog.Warningf("Node %q does not exist, vsphere CP will assume all disks %v are not attached to it.", nodeName, volPaths)
+				// make all the disks as detached and return false without error.
+				attached := make(map[string]bool)
+				for _, volPath := range volPaths {
+					attached[volPath] = false
+				}
+				return attached, nil
+			}
 			glog.Errorf("Failed to get VM object for node: %q. err: +%v", vSphereInstance, err)
 			return nil, err
 		}
-		nodeExist, err := vm.Exists(ctx)
```

Comment:
I have one more question. If node is powered off and the pod is placed on to other available nodes, will the volumes attached to powered off node detach automatically?

What logic in here, would cause it to happen? 

Its because i have one PR which does this https://github.com/kubernetes/kubernetes/pull/40118. Throwing an error when node is powered off caused the volume to be detached automatically.